machine.go

```diff
@@ -0,0 +1,205 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package gomaasapi
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/schema"
+	"github.com/juju/version"
+)
+
+type machine struct {
+	// Add the controller in when we need to do things with the machine.
+	// controller Controller
+	resourceURI string
+
+	systemID    string
+	hostname    string
+	fqdn        string
+	ipAddresses []string
+	memory      int
+	cpuCount    int
+	powerState  string
+
+	zone *zone
+
+	operatingSystem string
+	distroSeries    string
+	architecture    string
+	// NOTE: status is an int, need to get status values from MAAS
+	// for now we'll set status to be the status_name
+	status string
+}
+
+// SystemId implements Machine.
+func (m *machine) SystemId() string {
+	return m.systemID
+}
+
+// Hostname implements Machine.
+func (m *machine) Hostname() string {
+	return m.hostname
+}
+
+// FQDN implements Machine.
+func (m *machine) FQDN() string {
+	return m.fqdn
+}
+
+// IPAddresses implements Machine.
+func (m *machine) IPAddresses() []string {
+	return m.ipAddresses
+}
+
+// Memory implements Machine.
+func (m *machine) Memory() int {
+	return m.memory
+}
+
+// CpuCount implements Machine.
+func (m *machine) CpuCount() int {
+	return m.cpuCount
+}
+
+// PowerState implements Machine.
+func (m *machine) PowerState() string {
+	return m.powerState
+}
+
+// Zone implements Machine.
+func (m *machine) Zone() Zone {
+	return m.zone
+}
+
+// OperatingSystem implements Machine.
+func (m *machine) OperatingSystem() string {
+	return m.operatingSystem
+}
+
+// DistroSeries implements Machine.
+func (m *machine) DistroSeries() string {
+	return m.distroSeries
+}
+
+// Architecture implements Machine.
+func (m *machine) Architecture() string {
+	return m.architecture
+}
+
+// Status implements Machine.
+func (m *machine) Status() string {
```

Comment:
StatusName ? Note that we'll also need status_message too, but we'll be adding new fields as we need them.